app/helpers/close_contact_query_helper.rb

```diff
@@ -5,4 +5,108 @@ module CloseContactQueryHelper
   def close_contacts_by_patient_ids(patient_ids)
     CloseContact.where(patient_id: patient_ids).order(:patient_id)
   end
+
+  # Validates the request params for the index action
+  def validate_close_contact_query(params)
+    permitted_params = params.permit(:entries, :page, :search, :order, :direction)
+
+    patient_id = params.require(:patient_id)&.to_i
+    search_text = permitted_params[:search]
+    sort_order = permitted_params[:order]
+    sort_direction = permitted_params[:direction]
+    # Assume default values if pagination data is not explicitly specified
+    entries = permitted_params[:entries]&.to_i || 10
+    page = permitted_params[:page]&.to_i || 0
+
+    if sort_direction.present? && !%w[asc desc].include?(sort_direction)
+      error_message = "Unable to sort column in specified direction in request: '#{sort_direction}'"
+      raise StandardError, error_message
+    end
+
+    if sort_order.present? && !%w[first_name last_name primary_telephone email last_date_of_exposure assigned_user contact_attempts enrolled_id
+                                  notes].include?(sort_order)
+      error_message = "Unable to sort by specified column in request: '#{sort_order}'"
+      raise StandardError, error_message
+    end
+
+    # Either both the sort order and direction are provided, or neither is provided
+    if (sort_direction.blank? && sort_order.present?) || (sort_direction.present? && sort_order.blank?)
+      error_message = 'Must have both a sort column and direction specified or neither specified. Requested column to sort: '\
+                      "'#{sort_order}'', with specified direction: '#{sort_direction}'"
+      raise StandardError, error_message
+    end
+
+    if entries.negative? || page.negative?
+      error_message = "Invalid pagination options. Number of entries: #{entries}. Page: #{page}"
```

Comment:
Can we create individual error messages for these fields that are more specific? similar to this: https://github.com/SaraAlert/SaraAlert/blob/master/app/helpers/patient_query_helper.rb#L14 (same for vaccine_query_helper)